tests/Controller/SnippetControllerTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Tests\Controller;
+
+use App\Service\SnippetGenerator;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class SnippetControllerTest extends WebTestCase
+{
+    /**
+     * @var Client
+     */
+    var $client;
+
+    /**
+     * @var SnippetGenerator
+     */
+    var $generator;
+
+    public function setUp()
+    {
+        $this->client = static::createClient();
+
+        $this->generator = $this->getMockBuilder(SnippetGenerator::class)
+                    ->disableOriginalConstructor()
+                    ->getMock();
+
+        static::$kernel->getContainer()->set(SnippetGenerator::class, $this->generator);
```

Comment:
The warning on the CI is the fault of this line.
A possible solution could be to make the public service, but I think there is a cleaner way to get the `SnippetGenerator` service without making it public.
@garak could help us!
Thanks!

Reply:
Instead of mock service inside test, create a Stub and declare it as a service in test env.

Reply:
Since this community experiment also aims to grow the community. Could you post an example or a link that explain how to implement this?
Thank you!

Reply:
SnippetGenerator should implement an interface. Such interface should be aliased to SnippetGenerator in services.yaml, and to SnippetGeneratorStub (implementing the same interface, of course) in services_test.yaml